src/praisonai-agents/praisonaiagents/background/runner.py

```
@@ -15,6 +15,24 @@
+    task: "BackgroundTask",
+) -> None:
+    """Invoke ``on_complete`` for a terminal task, swallowing callback errors.
+
+    Shared by every terminal branch (success, failure, timeout, cancellation)
+    so the completion contract holds regardless of how the task ended.
+    """
+    if on_complete is None:
+        return
+    try:
+        on_complete(task)
+    except Exception as cb_error:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)